src/components/panel/panel.js

```diff
@@ -215,19 +213,66 @@ angular
  * @ngdoc method
  * @name MdPanelRef#open
  * @description
- * If the panel is not visible, opens an already created and configured panel.
+ * Attaches and shows the panel.
  *
- * @returns {angular.$q.Promise} A promise that is resolved when the panel is
- * closed.
+ * @returns {!angular.$q.Promise} A promise that is resolved when the panel is
+ * opened.
  */
 
 /**
  * @ngdoc method
  * @name MdPanelRef#close
  * @description
- * If the panel is visible, closes the panel, resolving the promise that is
- * returned from `MdPanelRef#open`. This method destroys the reference to the
- * panel. In order to open the panel again, a new one must be created.
+ * Hides and detaches the panel. This method destroys the reference to the panel.
+ * In order to open the panel again, a new one must be created.
+ *
+ * @returns {!angular.$q.Promise} A promise that is resolved when the panel is
+ * closed.
+ */
+
+/**
+ * @ngdoc method
+ * @name MdPanelRef#attachOnly
+ * @description
+ * Create the panel elements and attach them to the DOM. The panel will be
+ * hidden by default. Call toggleVisibility() or setVisible(true) to make the
+ * panel visible.
+ *
+ * @returns {!angular.$q.Promise} A promise that is resolved when the panel is
+ * attached.
+ */
+
+/**
+ * @ngdoc method
+ * @name MdPanelRef#detach
+ * @description
+ * Removes the panel from the DOM. If the panel is visible, hide the panel,
+ * then detach.
+ *
+ * @returns {!angular.$q.Promise} A promise that is resolved when the panel is
+ * detached.
+ */
+
+/**
+ * @ngdoc method
+ * @name MdPanelRef#setVisibililty
```

Comment:
To be clear, I mean adding `addClass` / `removeClass` / `toggleClass` to `MdPanelRef`